File: src/main/java/net/onelitefeather/pandorascluster/enums/Permission.java
package net.onelitefeather.pandorascluster.enums;

import net.onelitefeather.pandorascluster.util.Constants;
import org.bukkit.permissions.Permissible;
import org.jetbrains.annotations.NotNull;

public enum Permission {

    BLOCK_BREAK("%s.unowned.block.break"),
    BLOCK_PLACE("%s.unowned.block.place"),
    ENTITY_MOUNT("%s.unowned.mount"),
    LAND_ENTRY_DENIED("%s.unowned.entry.denied"),

    PVP("%s.unowned.attack.player"),
    PVE("%s.unowned.attack.entity"),

    VEHICLE_ENTER("%s.unowned.vehicle.use"),
    VEHICLE_DAMAGE("%s.unowned.vehicle.damage"),
    VEHICLE_DESTROY("%s.unowned.vehicle.destroy"),
    EXPLOSION("%s.unowned.explosion"),
    INTERACT_CONTAINERS("%s.unowned.interact.container"),
    INTERACT_FARMLAND("%s.unowned.interact.farmland"),
    INTERACT_USE("%s.unowned.interact.use"),
    POTION_SPLASH("%s.unowned.potion.splash"),
    PROJECTILE_HIT_ENTITY("%s.unowned.projectile.hit");

    private final String permissionNode;

    Permission(@NotNull String permissionNode) {
        this.permissionNode = permissionNode;
    }

    @NotNull
    public String getPermissionNode() {
        return String.format(permissionNode, Constants.PLUGIN_NAME);
    }

    public boolean hasPermission(@NotNull Permissible permissible) {
       return permissible.hasPermission(getPermissionNode());
    }
}

File: src/main/kotlin/net/onelitefeather/pandorascluster/enums/Permission.kt
package net.onelitefeather.pandorascluster.enums

import org.bukkit.permissions.Permissible

enum class Permission(val permissionNode: String) {

    NO_CLAIM_LIMIT("pandorascluster.unlimit.claim"),
    FLAG_PERMISSION("pandorascluster.flags.flag.%s"),
    CLAIM_LIMIT("pandorascluster.limit.claim"),
    SET_LAND_ROLE("pandorascluster.admin.set.role"),
    SET_LAND_FLAG("pandorascluster.admin.set.flags"),
    SET_LAND_HOME("pandorascluster.admin.set.home"),
    SET_LAND_OWNER("pandorascluster.admin.set.owner"),
    UNOWNED_CHUNK("pandorascluster.unowned.access"),
    BLOCK_BREAK("pandorascluster.unowned.block.break"),
    BLOCK_PLACE("pandorascluster.unowned.block.place"),
    OWNED_CHUNK("pandorascluster.owned.access"),
    LAND_ENTRY_DENIED("pandorascluster.owned.entry.denied"),
    INTERACT_CONTAINERS("pandorascluster.owned.interact.container"),
    INTERACT_FARMLAND("pandorascluster.unowned.interact.farmland");

    fun hasPermission(permissible: Permissible): Boolean {
        return permissible.hasPermission(permissionNode)
    }
}